Introduce `Settings::Base` (#14138)
forem/forem: docs/backend/configuration.md

secrets easily through the application rather than having to mess with ENV
files. If you would like to try out Vault, follow our
[installation guide for setting it up locally](/installation/vault).

## `Settings::Base`

> This is our customized version of the
> [rails-settings-cached](https://github.com/huacnlee/rails-settings-cached) code
> which allows for multiple settings model by defining class-specic request
> caches.

Reviewer: Suggestion: rephrase this to avoid giving the impression we've forked or are somehow tied to `rails-setting-cached` (which is how I read it). 

```suggestion
This class is adapted from 
[rails-settings-cached](https://github.com/huacnlee/rails-settings-cached).
It allows for multiple settings model by defining class-specic request
caches.
```

Author: That's a good point. I actually already have some other changes locally so I won't commit the suggestion as-is but thanks for the feedback @joshpuetz!
